### ISSUE TITLE
Can ImageAI use newer Libraries?

### ISSUE THREAD

**Issue by damies13** [OlafenwaMoses/ImageAI#761]
Hi I was wanting to try ImageAI but was not able to get it working following the [Install Documentation](https://imageai.readthedocs.io/en/latest/#installing-imageai)

```
pip install tensorflow==2.4.0
```

The first issue was tensorflow 2.4.0 is no longer available so I just installed the latest version (2.9.1), I also tried installing the oldest available version 2.5.0

I also had the same issue with
```
pip install keras==2.4.3 numpy==1.19.3 pillow==7.0.0 scipy==1.4.1 h5py==2.10.0 matplotlib==3.3.2 opencv-python keras-resnet==0.2.0
```

For some of these libraries these version weren't available, I started as with tensorflow with the current version, then when I rolled back to tensorflow 2.5.0 I rolled these back to the closest available version

For Reference I'm using a vanilla Debian 11 install with Python 3.9.2

Other issues I had (that might be worth including in the documentation):
1. The dependancy on `libGL.so.1` was missed, so I needed to figure out that I needed to install
```
apt install libgl1-mesa-glx
```
2. tensorflow needs a lot of memory to install (>2GB), for my application I intend to use small ARM SBC's these often only have 512MB of ram so I set my VM up with the same specifications, however the tensorflow install just crashed, I needed to give more than 2GB of swap memory in order for tensorflow to install at all.

My main question is: Will there be a version soon of ImageAI that works with a more recent version of tensorflow and the other dependancies?

Dave.

**Comment by alonsoir** [comment]
This issue is quite important because right now anyone who doesn't have the dependencies saved by chance on their computer won't be able to configure the framework. I understand the difficulty of maintaining the software with the new versions of the dependencies, but it is absolutely necessary to keep the project alive.

**Comment by YakFromStateFarm** [comment]
I agree with the above and am willing to offer any help needed if that is of any interest.

**Comment by immortal-blog** [comment]
I agree. I cannot install and run directly because it uses too many old libs that don't exist on pip.
I don't want to find the old versions and install them one by one.
I just give up.

**Comment by tankxiaodi** [comment]
You might need to check the tensorflow-2.7.2 branch.
